scarb/src/ops/fmt.rs

```diff
@@ -42,28 +45,97 @@ pub fn format(opts: FmtOptions, ws: &Workspace<'_>) -> Result<bool> {
 
     let all_correct = AtomicBool::new(true);
 
+    if let Some(target_path) = &opts.target_path {
+        if target_path.is_file() {
+            format_single_file(target_path, &opts, ws, &all_correct)?;
+            return Ok(all_correct.load(Ordering::Acquire));
+        }
+        let pkg = ws
+            .members()
+            .find(|member| target_path.starts_with(member.root()));
+
+        if let Some(pkg) = pkg {
+            format_package(&pkg, &opts, ws, &all_correct)?;
+            return Ok(all_correct.load(Ordering::Acquire));
+        } else {
+            return Err(anyhow::anyhow!(
+                "File {:?} is not part of the workspace.",
+                target_path
+            ));
+        };
+    };
     for package_id in opts.packages.iter() {
         let package = ws.fetch_package(package_id)?;
+        format_package(package, &opts, ws, &all_correct)?;
+    }
 
-        let mut config = FormatterConfig::default();
-        if let Some(overrides) = package.tool_metadata("fmt") {
-            config = toml_merge(&config, overrides)?;
-        }
-        let fmt = CairoFormatter::new(config);
+    Ok(all_correct.load(Ordering::Acquire))
```

Comment:
is this the right way of using `all_correct`?

Reply:
Seems alright to me :)